sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -56,16 +58,26 @@ case class BatchScanExec(
 
   @transient private lazy val filteredPartitions: Seq[Seq[InputPartition]] = {
     val dataSourceFilters = runtimeFilters.flatMap {
-      case DynamicPruningExpression(e) => DataSourceStrategy.translateRuntimeFilter(e)
+      case DynamicPruningExpression(e) =>
+        scan match {
+          case _: SupportsRuntimeFiltering =>
+            DataSourceStrategy.translateRuntimeFilter(e)
+          case _: SupportsRuntimeV2Filtering =>
```

Comment:
shall we make `SupportsRuntimeV2Filtering` have higher priority over `SupportsRuntimeFiltering`? Also we need to document the behavior if a source implements both of them

Reply:
It doesn't seem to me that a data source would implement both `SupportsRuntimeV2Filtering` and `SupportsRuntimeFiltering`?